### DISCUSSION TITLE
Strings - StringBuilder [Fundamentos C#]

### DISCUSSION THREAD

**Discussion by leorms11** [balta-io/2801#82] [Perguntas e Respostas] [answered]
Olá, estou com uma dúvida com relação ao instanciar um objeto, por exemplo na aula sobre escrita em arquivos, foi utilizado o using (var file = new StreamWrite(...)) { } porém na aula sobre o StringBuilder quando utilizado o var texto = new StringBuilder(), não foi utilizado o using. A dúvida é, como saber quando o using será necessário?

**Comment by andrebaltieri** [comment, marked as answer]
O `using` faz o `dispose` (destrói) do objeto no fim do uso.
Então para casos como escrita de arquivo, precisamos dele para garantir que no término da execução do processo, tudo vai ser destruído e o arquivo não vai ficar aberto, impedindo que outros programas o consumam por exemplo.

O mesmo acontece com conexões a banco de dados e outros recursos que precisam ser explicitamente temporários.
